### DISCUSSION TITLE
Feature Request: Arch Linux (SteamOS)

### DISCUSSION THREAD

**Discussion by CorruptName** [svpcom/wfb-ng#268] [General]
Hello, I would really like to try to test out WFB-ng on SteamDeck. Would be cool to use steamdeck as controller for FPV.
Is there any information on installing under Arch Linux currently available?
Thanks

**Comment by svpcom** [comment]
https://github.com/svpcom/wfb-ng/wiki/Setup-HOWTO

**Comment by 1anakin20** [comment]
A few notes on how I built it for Arch, Manjaro. I installed each dependency Arch equivalent mentioned in
```
sudo apt install python3-all libpcap-dev libsodium-dev libevent-dev python3-pip python3-pyroute2 \
            python3-twisted python3-serial python3-all-dev iw socat virtualenv \
            debhelper dh-python fakeroot build-essential python3-msgpack \
            python3-setuptools libgstrtspserver-1.0-dev -y
```

Debian uses a package named `trail3` while in Arch it is `trail`.
There are two options, either make a symbolic link from `trail` to `trail3` or change it to `trail` in the Makefile.

Compile it with
```
make
```

For some reason, `make` kept failing because it did not find a `wfb_rtsp` file. So I called stage directly `make wfb_rtsp`. Might just be a me issue. 

I had to set the following environment variables to run setup.py.
```
export VERSION=$(python version.py $(git show -s --format="%ct" HEAD) stable)
export COMMIT=$(git rev-parse HEAD)
export SOURCE_DATE_EPOCH=$(git show -s --format="%ct" HEAD)
```

To run the setup.py, I had to install additional python modules trough pacman. This might not be an exhaustive list of them.
```
sudo pacman -S python-setuptools
sudo pacman -S python-msgpack
sudo pacman -S python-msgpack
sudo pacman -S python-pyroute2
sudo pacman -S python-virtualenv
sudo pacman -S python-pyyaml
sudo pacman -S python-jinja
```

Then  run the setup. `-E` to use the environment variables set earlier.
```
sudo -E python setup.py install
```

**Reply by svpcom** [reply]
Have you used master or release branch? Issue with environment variables was recently fixed in master branch.
Thanks for reported issues with trial3 and wfb_rtsp. I've fixed them.
